pip/index.py

```diff
@@ -309,7 +309,7 @@ def mkurl_pypi_url(url):
                 loc = loc + '/'
             return loc
 
-        url_name = req.url_name
+        project_url_name = urllib_parse.quote(project_name.lower())
```

Comment:
how bout refactor this into new a util function, and have this location and `req.url_name` call that, otherwise we're duplicating this, and they could diverge.